src/utils/yamlUtils.ts

```diff
@@ -0,0 +1,72 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { readFile } from "fs-extra";
+import { basename } from "path";
+import { ext } from "vscode-azureappservice/out/src/extensionVariables";
+import { parse } from "yaml";
+import { BuildConfig, BuildConfigs } from "../tree/localProject/ConfigGroupTreeItem";
+import { localize } from "./localize";
+
+type BuildDeployStep = {
+    id: 'builddeploy',
+    with: {
+        api_location: string,
+        app_location: string,
+        output_location?: string,
+        app_artifact_location?: string
+    }
+}
+
+export async function parseYamlFile(yamlFilePath: string): Promise<BuildConfigs | undefined> {
+    const contents: string = (await readFile(yamlFilePath)).toString();
+    const buildDeployStep: BuildDeployStep | undefined = await getBuildDeployStep(contents, basename(yamlFilePath));
+
+    if (buildDeployStep) {
+        return {
+            api_location: buildDeployStep.with.api_location,
+            app_location: buildDeployStep.with.app_location,
+            output_location: buildDeployStep.with.output_location,
+            app_artifact_location: buildDeployStep.with.app_artifact_location
+        };
+    }
+
+    return undefined;
+}
+
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins */
+async function getBuildDeployStep(yamlFileContents: string, yamlFileName: string): Promise<BuildDeployStep | undefined> {
+    if (/Azure\/static-web-apps-deploy/.test(yamlFileContents)) {
+        const parsedYaml: any = await parse(yamlFileContents);
+
+        for (const job of <any[]>Object.values(parsedYaml.jobs)) {
+            for (const step of <any[]>Object.values(job['steps'])) {
+                if (step.id === 'builddeploy' && step.with) {
+                    if (!stepIncludesBuildConfig(step, 'app_location')) {
+                        return showYamlWarningMessage(yamlFileName, 'app_location');
```

Comment:
I think it's pretty unintuitive that `showYamlWarningMessage` returns `undefined`.  I understand why you're doing it though.

I hatched this as a solution:

```
if (stepIncludesBuildConfig(step, 'app_location')) {
                                if (stepIncludesBuildConfig(step, 'api_location')) {
                                    if (stepIncludesBuildConfig(step, 'output_location') || stepIncludesBuildConfig(step, 'app_artifact_location')) {
                                        return <BuildDeployStep>step;
                                    }
                                    void showYamlWarningMessage(yamlFileName, 'output_location');
                                }
                                void showYamlWarningMessage(yamlFileName, 'api_location');
                            }
                            void showYamlWarningMessage(yamlFileName, 'app_location');
                        }
```

But it honestly might be grosser/less intuitive anyway.

Reply:
Unfortunately this shows warnings for buildConfigs that are already defined. For instance if `app_location` is defined but `api_location` isn't, these warnings are shown:

![image](https://user-images.githubusercontent.com/22795803/113631876-f9427700-961e-11eb-930e-b1aee63d0f10.png)

It just occurred to me that we could throw elses in there to fix that problem but I just committed changes to make the old version more readable. Let me know which version looks better to ya - I don't have much of a preference

Reply:
I like your solution.  The main thing was relying on returning `showYamlWarningMessage` which I thought was kind of odd.